library/proc_macro/src/lib.rs

```diff
@@ -880,7 +880,47 @@ impl Group {
     /// tokens at the level of the `Group`.
     #[stable(feature = "proc_macro_lib2", since = "1.29.0")]
     pub fn set_span(&mut self, span: Span) {
-        self.0.span = bridge::DelimSpan::from_single(span.0);
+        let (open_chr, close_chr) = match self.0.delimiter {
+            Delimiter::Brace => (b'{', b'}'),
+            Delimiter::Bracket => (b'[', b']'),
+            Delimiter::Parenthesis => (b'(', b')'),
+            Delimiter::None => {
+                // None delimiters are the whole span for open and close
+                // same behavior as in the compiler.
+                self.0.span = bridge::DelimSpan::from_single(span.0);
+                return;
+            }
+        };
+
+        // No source text to check or source text is too small.
+        // just go back to default behavior.
+        let source = match span.0.source_text() {
```

Comment:
This seems like quite the expensive operation to call here. For one this goes through the bridge (which is expensive for rust-analyzer, especially this call, as once we implement it will always do an RPC roundtrip), but that aside, this also always allocates a new string.

Reply:
>to at least to make sure the spans are from the same file

That also requires going through the bridge though.